plutus-core/plutus-core/src/PlutusCore/Core/Type.hs

```diff
@@ -246,6 +246,7 @@ mapFun f = go where
 -- This marking allows us to skip the (de)serialization of binders at LamAbs/TyAbs positions
 -- iff 'name' is DeBruijn-encoded (level or index). See for example the instance of  'UntypedPlutusCore.Core.Instance.Flat'
 newtype Binder name = Binder { unBinder :: name }
+    deriving stock (Eq, Show, Functor)
```

Comment:
It's not really important, but please make it

```haskell
newtype Binder name = Binder { unBinder :: name }
    deriving stock (Eq, Show)

instance Functor Binder where
    fmap = coerce
    {-# INLINE fmap #-}
```

because apparently GHC generates unoptimal code (see the slack [discussion](https://input-output-rnd.slack.com/archives/C48HA39RS/p1687917638566799)).

Reply:
I see. Still I am pondering between using your optimized fmap or just inline the coerce itself. It's just that `coerce` bit us before, so I was very reluctant to use it again. 

Note to self : `coerce` without type applications/signatures is dangerous!

Reply:
What do you think about still deriving stock Functor but then using `coerceVia fmap`?

Like:

```
    size = size . coerceVia fmap fromFake
    encode = encode . coerceVia fmap fromFake
    decode =  coerceVia fmap toFake <$> decode
```

Reply:
You just derive `Functor` via `Identity` and that is simpler than calling `coerceVia` at the call site.

But I don't mind using `coerceVia` there or just `fmap` directly, performance should be identical.

Reply:
I just saw the thread, I will do that.